lib/Doctrine/ORM/Internal/Hydration/IterableResult.php

```diff
@@ -65,7 +65,7 @@ public function next()
     }
 
     /**
-     * @return mixed
+     * @return mixed[]|false|null
```

Comment:
`false` is not a type, it should be `bool` instead

Reply:
If we have `bool`, then analysis of the `foreach` is confused by a possibly `true` value for `$row`:

```php
function foo(IterableResult $i) {
  $row = $i->current();
  if ($row) {
    foreach ($row as $column) {}
  }
}

Reply:
Ok, keep false, we can revisit in future.